src/internal.h:59:25: error: sys/utsname.h: No such file or directory

MinGW, ver.4.4.5

i tried it after merge pull request #443 from cesanta/uname_fix

src/internal.h:80:25: error: sys/utsname.h: No such file or directory

the same problem, but other string number

--- a/src/internal.h
+++ b/src/internal.h
@@ -11,7 +11,6 @@
 /* Check whether we're compiling in an environment with no filesystem */
 #if defined(ARDUINO) && (ARDUINO == 106)
 #define V7_NO_FS
-#define V7_NO_POSIX
 #endif
 
 /*
@@ -56,7 +55,6 @@
 #include "../v7.h"
 
 #ifdef V7_WINDOWS
-#define V7_NO_POSIX
 #define vsnprintf _vsnprintf
 #define snprintf _snprintf
 #define isnan(x) _isnan(x)
@@ -74,10 +72,6 @@ typedef unsigned long uintptr_t;
 #include <stdint.h>
 #include <unistd.h>
 #include <fcntl.h>
-#endif
-
-#ifndef V7_NO_POSIX
-#include <sys/utsname.h>
 #endif
 
 /* Private API */

--- a/src/os.c
+++ b/src/os.c
@@ -104,39 +104,6 @@ static val_t OS_rename(struct v7 *v7, val_t this_obj, val_t args) {
 }
 #endif
 
-#ifndef V7_NO_POSIX
-static val_t OS_uname(struct v7 *v7, val_t this_obj, val_t args) {
-  int res = -1;
-  struct utsname name;
-  val_t ret = v7_create_object(v7);
-
-  (void) this_obj;
-  (void) args;
-  res = uname(&name);
-
-  v7_set_property(v7, ret, "errno", 5, 0,
-                  v7_create_number(res >= 0 ? 0 : errno));
-  if (res >= 0) {
-    v7_set_property(
-        v7, ret, "sysname", 7, 0,
-        v7_create_string(v7, name.sysname, strlen(name.sysname), 1));
-    v7_set_property(
-        v7, ret, "nodename", 8, 0,
-        v7_create_string(v7, name.nodename, strlen(name.nodename), 1));
-    v7_set_property(
-        v7, ret, "release", 7, 0,
-        v7_create_string(v7, name.release, strlen(name.release), 1));
-    v7_set_property(
-        v7, ret, "version", 7, 0,
-        v7_create_string(v7, name.version, strlen(name.version), 1));
-    v7_set_property(
-        v7, ret, "machine", 7, 0,
-        v7_create_string(v7, name.machine, strlen(name.machine), 1));
-  }
-  return ret;
-}
-#endif
-
 V7_PRIVATE void init_os(struct v7 *v7) {
   val_t os_obj = v7_create_object(v7);
   v7_set_property(v7, v7->global_object, "OS", 2, 0, os_obj);
@@ -147,8 +114,5 @@ V7_PRIVATE void init_os(struct v7 *v7) {
   set_cfunc_obj_prop(v7, os_obj, "write", OS_write);
   set_cfunc_obj_prop(v7, os_obj, "remove", OS_remove);
   set_cfunc_obj_prop(v7, os_obj, "rename", OS_rename);
-#endif
-#ifndef V7_NO_POSIX
-  set_cfunc_obj_prop(v7, os_obj, "uname", OS_uname);
 #endif
 }

--- a/tests/unit_test.c
+++ b/tests/unit_test.c
@@ -2129,29 +2129,6 @@ static const char *test_file(void) {
 }
 #endif
 
-#ifndef V7_NO_POSIX
-static const char *test_os(void) {
-  v7_val_t v;
-  struct v7 *v7 = v7_create();
-
-  ASSERT(v7_exec(v7, &v, "var name = OS.uname()") == V7_OK);
-  ASSERT(v7_exec(v7, &v, "name.errno") == V7_OK);
-  ASSERT(check_value(v7, v, "0"));
-  ASSERT(v7_exec(v7, &v, "typeof name.sysname") == V7_OK);
-  ASSERT(check_value(v7, v, "\"string\""));
-  ASSERT(v7_exec(v7, &v, "typeof name.nodename") == V7_OK);
-  ASSERT(check_value(v7, v, "\"string\""));
-  ASSERT(v7_exec(v7, &v, "typeof name.release") == V7_OK);
-  ASSERT(check_value(v7, v, "\"string\""));
-  ASSERT(v7_exec(v7, &v, "typeof name.version") == V7_OK);
-  ASSERT(check_value(v7, v, "\"string\""));
-  ASSERT(v7_exec(v7, &v, "typeof name.machine") == V7_OK);
-  ASSERT(check_value(v7, v, "\"string\""));
-
-  return NULL;
-}
-#endif
-
 static const char *run_all_tests(const char *filter) {
   RUN_TEST(test_unescape);
   RUN_TEST(test_to_json);
@@ -2162,9 +2139,6 @@ static const char *run_all_tests(const char *filter) {
   RUN_TEST(test_native_functions);
 #ifndef V7_NO_FS
   RUN_TEST(test_file);
-#endif
-#ifndef V7_NO_POSIX
-  RUN_TEST(test_os);
 #endif
   RUN_TEST(test_stdlib);
   RUN_TEST(test_runtime);

--- a/v7.c
+++ b/v7.c
@@ -645,7 +645,6 @@ struct gc_arena {
 /* Check whether we're compiling in an environment with no filesystem */
 #if defined(ARDUINO) && (ARDUINO == 106)
 #define V7_NO_FS
-#define V7_NO_POSIX
 #endif
 
 /*
@@ -689,7 +688,6 @@ struct gc_arena {
 /* Public API. Implemented in api.c */
 
 #ifdef V7_WINDOWS
-#define V7_NO_POSIX
 #define vsnprintf _vsnprintf
 #define snprintf _snprintf
 #define isnan(x) _isnan(x)
@@ -707,10 +705,6 @@ typedef unsigned long uintptr_t;
 #include <stdint.h>
 #include <unistd.h>
 #include <fcntl.h>
-#endif
-
-#ifndef V7_NO_POSIX
-#include <sys/utsname.h>
 #endif
 
 /* Private API */
@@ -14350,39 +14344,6 @@ static val_t OS_rename(struct v7 *v7, val_t this_obj, val_t args) {
 }
 #endif
 
-#ifndef V7_NO_POSIX
-static val_t OS_uname(struct v7 *v7, val_t this_obj, val_t args) {
-  int res = -1;
-  struct utsname name;
-  val_t ret = v7_create_object(v7);
-
-  (void) this_obj;
-  (void) args;
-  res = uname(&name);
-
-  v7_set_property(v7, ret, "errno", 5, 0,
-                  v7_create_number(res >= 0 ? 0 : errno));
-  if (res >= 0) {
-    v7_set_property(
-        v7, ret, "sysname", 7, 0,
-        v7_create_string(v7, name.sysname, strlen(name.sysname), 1));
-    v7_set_property(
-        v7, ret, "nodename", 8, 0,
-        v7_create_string(v7, name.nodename, strlen(name.nodename), 1));
-    v7_set_property(
-        v7, ret, "release", 7, 0,
-        v7_create_string(v7, name.release, strlen(name.release), 1));
-    v7_set_property(
-        v7, ret, "version", 7, 0,
-        v7_create_string(v7, name.version, strlen(name.version), 1));
-    v7_set_property(
-        v7, ret, "machine", 7, 0,
-        v7_create_string(v7, name.machine, strlen(name.machine), 1));
-  }
-  return ret;
-}
-#endif
-
 V7_PRIVATE void init_os(struct v7 *v7) {
   val_t os_obj = v7_create_object(v7);
   v7_set_property(v7, v7->global_object, "OS", 2, 0, os_obj);
@@ -14393,8 +14354,5 @@ V7_PRIVATE void init_os(struct v7 *v7) {
   set_cfunc_obj_prop(v7, os_obj, "write", OS_write);
   set_cfunc_obj_prop(v7, os_obj, "remove", OS_remove);
   set_cfunc_obj_prop(v7, os_obj, "rename", OS_rename);
-#endif
-#ifndef V7_NO_POSIX
-  set_cfunc_obj_prop(v7, os_obj, "uname", OS_uname);
 #endif
 }

